corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -587,12 +587,18 @@ def save_deprecated_form(form):
 
     @staticmethod
     @transaction.atomic
-    def update_form_problem_and_state(form):
-        with get_cursor(XFormInstanceSQL) as cursor:
-            cursor.execute(
-                'SELECT update_form_problem_and_state(%s, %s, %s)',
-                [form.form_id, form.problem, form.state]
-            )
+    def update_form(form):
+        from corehq.form_processor.change_publishers import publish_form_saved
+        from corehq.sql_db.util import get_db_alias_for_partitioned_doc
+        db_name = get_db_alias_for_partitioned_doc(form.form_id)
+        if not form.is_saved():
+            FormAccessorSQL.save_new_form(form)
```

Comment:
So that means the method shouldn't supposed unsaved forms, and I should raise an assertion if otherwise, yeah?